src/list.cpp

```diff
@@ -457,14 +457,11 @@ Uint32 list_Index(node_t* node)
 
 node_t* list_Node(list_t* list, int index)
 {
-	if ( index < 0 || index >= list_Size(list) )
-	{
+	if (index < 0) {
```

Comment:
The old method was just inefficient, this function still returns nullptr if index is beyond the list size.

the problem was that list_Size() walks the entire list to return the size which is redundant because we're already walking the list.

After the for statement is completed, `node` will be nullptr because `node->next` will be nullptr on the last node. So after for () completes, node will always be either the correct node or nullptr if the node was not found.